typebinder/src/exporters/check.rs

```diff
@@ -0,0 +1,133 @@
+use crate::exporters::utils::{get_file_contents, get_output_file_path};
+use crate::{
+    error::TsExportError,
+    exporters::{Exporter, HeaderComment},
+    pipeline::module_step::ModuleStepResultData,
+};
+use displaythis::Display;
+use std::collections::HashMap;
+use std::io::Read;
+use std::path::PathBuf;
+
+pub struct CheckExport {
+    root_path: PathBuf,
+    default_module_name: Option<String>,
+    header_comment: HeaderComment,
+    patches: HashMap<PathBuf, DiffPatch>,
+}
+
+impl Default for CheckExport {
+    fn default() -> Self {
+        let root_path = std::env::current_dir().expect("Failed to find current dir");
+        CheckExport {
+            root_path,
+            default_module_name: None,
+            header_comment: HeaderComment::Standard,
+            patches: HashMap::default(),
+        }
+    }
+}
+
+#[derive(Debug, Display)]
+pub enum DiffKind {
+    #[display("+{0}")]
+    Added(String),
+    #[display("-{0}")]
+    Removed(String),
+}
+
+#[derive(Debug, Display)]
+#[display("{line}: {kind}")]
+pub struct DiffChange {
+    line: usize,
+    kind: DiffKind,
+}
+
+#[derive(Debug)]
+pub enum DiffPatch {
+    NewFile(String),
+    Changes(Vec<DiffChange>),
+}
+
+pub fn display_patch((path, diff): (&PathBuf, &DiffPatch)) {
```

Comment:
I guess the reason for this is to be able to call
```
self.patches.iter().for_each(display_patch);
```
But I would still suggest introducing a new struct instead of tuple and do a little conversion. It would bring slightly more clarity. Or alternatively you can use 2 arguments instead of a tuple argument.